package/MDAnalysis/analysis/base.py

```diff
@@ -261,33 +319,57 @@ def __init__(self, function, trajectory=None, *args, **kwargs):
         super(AnalysisFromFunction, self).__init__(trajectory)
 
     def _prepare(self):
-        self.results = []
+        self.results.timeseries = []
 
     def _single_frame(self):
-        self.results.append(self.function(*self.args, **self.kwargs))
+        self.results.timeseries.append(self.function(*self.args, 
+                                                     **self.kwargs))
 
     def _conclude(self):
-        self.results = np.asarray(self.results)
+        self.results.frames = self.frames
+        self.results.times = self.times
+        self.results.timeseries = np.asarray(self.results.timeseries)
 
 
 def analysis_class(function):
-    """
-    Transform a function operating on a single frame to an analysis class
+    r"""Transform a function operating on a single frame to an 
+    :class:`AnalysisBase` class.
+
+    For an usage in a library we recommend the following style
```

Comment:
For use in the library ...

Please move to Example section at the end of the docs.

(I now realize that you've been following the existing docs, which probably predate our adherence to numpydocs formatting style.)